
+++
date = 2014-09-22T21:35:34Z
draft = false
title = "Primate DUF1220 analysis"
description = "Analysis of the evolution of the DUF1220 domain"
+++

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Introduction](#introduction)
- [Method summary](#method-summary)
- [Command history](#command-history)
  - [Primate sequences used](#primate-sequences-used)
  - [DUF1220 proteome annotation](#duf1220-proteome-annotation)
  - [Building nucleotide alignments of DUF1220 from protein domain annotation](#building-nucleotide-alignments-of-duf1220-from-protein-domain-annotation)
  - [Verification of domains using cDNA](#verification-of-domains-using-cdna)
  - [Pseudogenes in humans](#pseudogenes-in-humans)
  - [Counting only DUF1220 domains that are close to the CM promoter](#counting-only-duf1220-domains-that-are-close-to-the-cm-promoter)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->


## Introduction
This document describes the methods and every command used to analyze the distribution of the DUF1220([PF06758](http://pfam.xfam.org/family/duf1220)) protein domains across various primates.
The analysis and pipeline is based on some of the methods described by [O'Bleness et al.](http://g3journal.org/content/2/9/977). The main difference to th O'Bleness paper is,
that BLAST has been replaced by nucleotide hidden markov models.

All the code is available [on Github](https://github.com/qfma/duf1220) and may be used to recreate the analysis.

## Method summary

In order to characterize the distribution of DUF1220 domains in NBPF genes, we obtained 12 primate genomes from Ensembl [REF] version 77 (*Homo sapiens*, *Pan troglodytes*, *Gorilla gorilla*, *Pongo abelii*, *Nomascus leucogenys*, *Macaca mulatta*, *Tarsius syrichta*, *Microcebus murinus*, *Otolemur garnettii*, *Callithrix jacchus*, *Papio anubis*, *Chlorocebus sabaeus*). Additonally, we included three genomes from Pre-Ensembl [REF] (*Macaca fasicularis*, *Papio hamadryas*, *Saimiri boliviensis*), the *Pan paniscus* genome [REF] and the *Daubentonia madagascariensis* genome [REF]. We used HMMER [REF] to build a Hidden Markov Model (HMM) from the DUF1220([PF06758](http://pfam.xfam.org/family/duf1220)) seed alignment stored in the [PFAM database](http://pfam.xfam.org/family/duf1220#tabview=tab3). All Ensembl proteomes were searched using the DUF1220 HMM (hmmsearch, E-value < 1e-10), providing a first estimate of the DUF1220 in these species. Because protein predictions are only available for the 12 Ensembl genomes, we extracted the corresponding cDNA regions of significant protein domain hits and aligned the cDNA using Mafft [REF]. In order to detect DUF1120 domains in DNA, we used the alignment of the cDNA DUF1220 to build a nucleotide HMM (nHMM). We confirmed that the nHMM was effective in detecting DUF1120 domains by searching the cDNA sequences of all 12 Ensembl species (nhmmer, E-value < 1e-10). The number of domains detected by the protein HMM and the nHMM are  similar (REF Table), except for the human dataset, where  a large set of untranslated pseudogenes contains DUF1220 domains.

Because we are primarily interested in those DUF1220 domains that are located in NBPF genes we extracted the CM (Conserved Mammal) promoter region as defined by [O'Bleness et al.](http://g3journal.org/content/2/9/977). The CM promoter is around 900bp long and is positioned either upstream of a NBPF gene or in the fourth intron. We extracted 900bp upstream of human genes NBPF4, NBPF6 and NBPF7 and aligned them using MAFFT (mafft --globalpair). This alignment was again used to build a nucleotide HMM for the CM promoter. In the next step, we searched all primate genomes using the CM promoter nHMM and extracted DNA 100kb , the approximate length of the longest NBPF gene in human, up- and downstream of significant CM promoter hits (nhmmer, E-value < 1e-10). The DUF1120 nHMM was then used to detect DUF1120 genes in the CM regions in all primates, ensuring that only those domains are counted that are located close to a CM promoter (Table REF.)

All scripts and data used in the analysis are made available on [Github](https://github.com/qfma/duf1220) under a MIT license.

## Command history and data used

### Primate sequences used

**[Ensembl](http://www.ensembl.org/index.html)** version 77:

1. Homo sapiens (Humans)
2. Pan troglodytes (Common Chimpanzees)
3. Gorilla gorilla (Gorilla)
4. Pongo abelii (Orangutan)
5. Nomascus leucogenys (Gibbon)
6. Macaca mulatta (Rhesus Macaque)
7. Tarsius syrichta (Tarsier)
8. Microcebus murinus (Mouse lemur)
9. Otolemur garnettii (Bushbaby)
10. Callithrix jacchus (Marmoset)
11. Papio anubis (Olive baboon)
12. Chlorocebus sabaeus (Green Vervet monkey)

**[Pre-Ensembl](http://pre.ensembl.org/index.html)**:

13. Macaca fasicularis (Crab-eating macaque)
14. Papio hamadryas (Hamadryas baboon)
15. Saimiri boliviensis (Squirrel monkey)


**George et al. exomes:*

**Note:**: were excluded from the final analysis

Paper is [here](http://genome.cshlp.org/content/21/10/1686.full) and data should be available [here](http://depts.washington.edu/swansonw/Swanson_Lab/Data.html)

**Note**: This requires a fucking proprietary piece of crappy software that kills my internet connection. Yes, it kills my router, my modem, my computer.

16. Chlorocebus aethiops (Grivet velvet monkey)
17. Colobus angolensis (Angola colobus)
18. Saguinus midas (red-handed tamarin)
19. Macaque (duplication Ensembl)

**MISC genomes:**

18. Homo neanderthalensis (Neanderthal) [Data download](http://neandertal.ensemblgenomes.org/index.html)

**Note:** Only BAM alignments?

19. Pan paniscus (Bonobo) [here](http://www.eva.mpg.de/bonobo-genome/data.html)

20. Daubentonia madagascariensis (Aye-aye): Paper is [here](http://gbe.oxfordjournals.org/content/4/2/126.long) data is [here](http://giladlab.uchicago.edu/data/AyeAyeGenome/)


### DUF1220 proteome annotation

The DUF1220([PF06758](http://pfam.xfam.org/family/duf1220)) seed alignment was downloaded from the PFAM database [here](http://pfam.xfam.org/family/duf1220#tabview=tab3).
The hidden markov model was built using HMMER 3.1b1.

``` bash
hmmbuild -n duf1220_seed --cpu 8  PF06758_seed.hmm PF06758_seed.txt
```

The PF06758_seed.hmm was used to search the proteomes of all 12 Ensembl primate proteomes using ```hmmsearch``` with the E-value threhsold of 1e-10.
Only the longest isoform was used for all proteomes.

``` bash
# Make longest isoforms
./get-longest-isoform.py sequences/pep sequences/pep
# Run hmmsearch against all proteomes
./hmmsearch-vs-pep-all-longest.sh
```
The output domtblout and tblout files were stored in ```2015-01-14-hmmsearch-vs-pep-all-longest```

``` bash
# Count the number of annotated domains
./get-peptide-domain-count.sh
Callithrix_jacchus: 14
Chlorocebus_sabaeus: 21
Gorilla_gorilla: 44
Homo_sapiens: 248
Macaca_mulatta: 23
Microcebus_murinus: 0
Nomascus_leucogenys: 5
Otolemur_garnettii: 0
Pan_troglodytes: 42
Papio_anubis: 17
Pongo_abelii: 20
Tarsius_syrichta: 8
```

### Building nucleotide alignments of DUF1220 from protein domain annotation

In order to detect DUF1220 in all species where protein translations are not available, we extracted the cDNA sequences that correspond to
the position of the DUF1220 domains in the proteomes. This information is extracted from the *.domtblout files.

``` bash
./get-all-positions.sh
```

The output is stored in ```2015-01-14-duf1220-all-ensembl-nucl```.  The cDNA fragments are aligned using ```mafft``` and this alignment is then used to create a hidden markove model with ```hmmbuild```.

### Verification of domains using cDNA

We now verified the DUF1220 counts using the CDNA of the 12 Ensembl species. The E-value cutoff is again 1e-10.

``` bash
# Run nhmmer
./nhmmer-vs-cdna-all-longest.sh
# Count the results
./get-cdna-domain-count.sh
Callithrix_jacchus: 15
Chlorocebus_sabaeus: 26
Gorilla_gorilla: 48
Homo_sapiens: 323
Macaca_mulatta: 25
Microcebus_murinus: 2
Nomascus_leucogenys: 7
Otolemur_garnettii: 2
Pan_troglodytes: 52
Papio_anubis: 21
Pongo_abelii: 25
Tarsius_syrichta: 4
```

### Pseudogenes in humans

The large number of domains found via cDNA can be explained through the presence of partly overlapping scaffolds that contain pseudogenes:

```text
ENST00000619932,ENSG00000272150,7,transcribed_unprocessed_pseudogene
ENST00000413192,ENSG00000240667,1,unprocessed_pseudogene
ENST00000617931,ENSG00000268043,14,protein_coding
ENST00000577412,ENSG00000266338,7,protein_coding
ENST00000609741,ENSG00000273136,11,protein_coding
ENST00000453025,ENSG00000227001,3,unprocessed_pseudogene
ENST00000318220,ENSG00000142794,6,protein_coding
ENST00000613157,ENSG00000196427,5,protein_coding
ENST00000612480,ENSG00000275131,1,unprocessed_pseudogene
ENST00000583271,ENSG00000270231,11,unprocessed_pseudogene
ENST00000369373,ENSG00000162825,68,protein_coding
ENST00000449715,ENSG00000203825,1,unprocessed_pseudogene
ENST00000615281,ENSG00000263956,9,protein_coding
ENST00000621744,ENSG00000271383,46,protein_coding
ENST00000612520,ENSG00000213240,7,protein_coding
ENST00000444082,ENSG00000227242,5,unprocessed_pseudogene
ENST00000425093,ENSG00000231382,2,unprocessed_pseudogene
ENST00000294652,ENSG00000186086,5,protein_coding
ENST00000357046,ENSG00000243967,3,transcribed_unprocessed_pseudogene
ENST00000445758,ENSG00000215864,3,unprocessed_pseudogene
ENST00000401087,ENSG00000179571,7,unprocessed_pseudogene
ENST00000583866,ENSG00000271425,44,protein_coding
ENST00000615421,ENSG00000269713,11,protein_coding
ENST00000430580,ENSG00000219481,10,protein_coding
ENST00000619423,ENSG00000270629,34,protein_coding
ENST00000590707,ENSG00000205449,2,transcribed_unprocessed_pseudogene
Protein_coding_domains,277
Pseudogene_domains,46
```

Statistics are stored in ```2015-01-14-nhmmer-vs-cdna-longest-stats``` and were generated with

```bash
./get-cdna-statistics.py 2015-01-14-nhmmer-vs-cdna-longest
```

### Counting only DUF1220 domains that are close to the CM promoter

As we are interested in DUF1220 counts across a broad pyhlogenetic range and no protein predictions are available for all genomes, we will use the nucleotide HMM to detect DUF1220 domains. Additionally, we are only interested in those DUF1220 domains that occur in NBPF genes. The NBPF genes are characterized by [O'Bleness et al.](http://g3journal.org/content/2/9/977) as those genes that contain a CM (conserved mammal promoter). The CM promoter is around 900bp long and is positioned either upstream of the gene or in the fourth intron. We extracted the 900 bp upstream for the human genes NBPF4, NBPF6 and NBPF7.

```bash
./get-CM-promoter-regions.sh
```

The promoter regions were aligned using ```mafft --globalpair``` and another nucleotide hmm was built using ```hmmbuild```.

This hidden markov model was then used to search the entire genomes of all primates to detect the location of this promoter. We then extracted the promoter region plus 100kb upstream and downstream for each genome and used only these regions to search for DUF1220 domains. This guarantees that only DUF1220 domains are counted that are located around the CM promoter and therefore likely part of NBPF genes. Cutoff is 1e-10.

```bash
# Search genomes for CM promoter
./nhmmer-CM-vs-dna.sh
# Extract DNA regions via Ensembl Rest API
./get-all-CM-regions-from-tblout.sh
# Manually extract DNA regions for non-Ensembl genomes
./get-all-CM-regions-from-tblout-fasta.sh
```

Regions stored in ```2015-01-14-all-CM-regions-plus-flank```
Now those DNA regions were searched using the DUF1220 hidden markov model
```bash
# Find DUF1220 domains in CM regions
./nhmmer-vs-CM-regions.sh
# Count the domains per species
./get-dna-domain-count.sh 2015-01-14-nhmmer-vs-CM-regions-plus-flank
Callithrix_jacchus: 78
Chlorocebus_sabaeus: 96
Deubetonia_madagascariensis: 0
Gorilla_gorilla: 90
Homo_sapiens: 317
Macaca_fascicularis: 79
Macaca_fascicularis_ex: 0
Macaca_mulatta: 79
Microcebus_murinus: 3
Nomascus_leucogenys: 53
Otolemur_garnettii: 13
Pan_paniscus: 30
Pan_paniscus_scaf: 0
Pan_troglodytes: 91
Papio_anubis: 97
Papio_hamadryas: 128
Pongo_abelii: 69
Saimiri_boliviensis: 2
Tarsius_syrichta: 24
```


<!--
Old DNA only results:

```bash
./nhmmer-vs-dna.sh
./get-dna-domain-count.sh
Callithrix_jacchus: 75
Chlorocebus_aethiops_ex: 0
Chlorocebus_sabaeus: 48
Colobus_angolensis_ex: 0
Deubetonia_madagascariensis: 26
Gorilla_gorilla: 97
Homo_sapiens: 295
Macaca_fascicularis: 67
Macaca_fascicularis_ex: 45
Macaca_mulatta: 75
Macaca_mulatta_ex: 43
Microcebus_murinus: 4
Nomascus_leucogenys: 59
Otolemur_garnettii: 7
Pan_paniscus: 34
Pan_troglodytes: 138
Papio_anubis: 76
Papio_hamadryas: 39
Pongo_abelii: 101
Saguinus_midas_ex: 0
Saimiri_boliviensis: 20
Tarsius_syrichta: 62
```
-->

